visionmetrics/prediction_filters.py

```diff
@@ -0,0 +1,102 @@
+import numpy as np
+from abc import ABC, abstractmethod
+
+
+class PredictionFilter(ABC):
```

Comment:
I'm not sure if we want to keep Filter class. If this is used by only one metrics, maybe it's simpler if we just implement the filter in the metrics class.